### DISCUSSION TITLE
Call to undefined function pg_connect()

### DISCUSSION THREAD

**Discussion by eadistar** [brefphp/bref#1137] [Support] [answered]
HI,

I'm getting  Call to undefined function pg_connect() on aws lambda.
I think this is related to extenstion of pdo_pgsql.

I'm using  "bref/bref": "0.5.x"

Does this version support php/conf.d/php.ini ? Does it support pdo_pgsql 
I'm using 
:layer:php-80
I have tried it but its not working.
Is there anything I'm missing?

**Comment by deleugpn** [comment, marked as answer]
Hi there! Welcome to Bref!

There are a couple of things to unpack from your issue. 1st, version 0.5 is a bit outdated and updating to version `^1.5` could help a bit (I don't remember if PHP 8 is compatible with bref 0.5 to be honest).

pg_connect() is undefined because it is only available as an additional extension. You can check the extensions here: https://bref.sh/docs/environment/php.html#extensions
If you pull in the pdo_pgsql extension into your Lambda, it should solve the issue. You will not need to modify `php.ini` as the extension layer will be doing that on your behalf.

**Reply by eadistar** [reply]
Thanks. I figured it out. due to some issues related latest bref version i'm not able to fulfilled my current requirement. 
As we are by default providing mysqli extension , I think we should also add pgsql extension which provide general pg_connect like functions.
I changed my code to PDO based to workaround.

**Reply by t-richard** [reply]
Even in 0.5, pdo_pgsql is provided by default but needs to be enabled from a custom php.ini (see https://github.com/brefphp/bref/blob/0.5.33/runtime/base/php-80.Dockerfile#L99)

Providing a custom ini file was also supported, create this file in your project `php/conf.d/php.ini` like showed [here](https://github.com/brefphp/bref/blob/0.5.33/docs/environment/php.md#customizing-phpini)

You should not need more to make it work.
